scipy/special/xsf/numpy.h

```diff
@@ -14,14 +14,37 @@
 
 #include <numpy/arrayobject.h>
 #include <numpy/npy_3kcompat.h>
+#include <numpy/npy_math.h>
 #include <numpy/ufuncobject.h>
 
 #include "dual.h"
-#include "sf_error.h"
+#include "error.h"
 #include "third_party/kokkos/mdspan.hpp"
 
 namespace xsf {
 namespace numpy {
+
+    // This is needed by sf_error, it is defined in the Cython "_ufuncs_extra_code_common.pxi" for "_generate_pyx.py".
+    // It exists to "call PyUFunc_getfperr in a context where PyUFunc_API array is initialized", but here we are
+    // already in such a context.
+    extern "C" int wrap_PyUFunc_getfperr() { return PyUFunc_getfperr(); }
```

Comment:
I'm not sure this function should be part of `xsf`, isn't this just SciPy-specific error handling? But I can understand if it needs to be here temporarily.